pkg/csi/service/vanilla/controller.go

```diff
@@ -379,7 +380,25 @@ func (c *controller) createBlockVolume(ctx context.Context, req *csi.CreateVolum
 			log.Errorf(errMsg)
 			return nil, status.Error(codes.NotFound, errMsg)
 		}
-		sharedDatastores, datastoreTopologyMap, err = c.nodeMgr.GetSharedDatastoresInTopology(ctx, topologyRequirement, c.manager.CnsConfig.Labels.Zone, c.manager.CnsConfig.Labels.Region)
+		vcenter, err := c.manager.VcenterManager.GetVirtualCenter(ctx, c.manager.VcenterConfig.Host)
+		if err != nil {
+			errMsg := fmt.Sprintf("Failed to get vCenter. Err: %v", err)
+			log.Errorf(errMsg)
+			return nil, status.Error(codes.NotFound, errMsg)
+		}
+		tagManager, err := cnsvsphere.GetTagManager(ctx, vcenter)
+		defer func() {
+			err = tagManager.Logout(ctx)
```

Comment:
If we get an error response from Line: 389, and tagManager is nil. Wouldn't this line cause NPE ?
I think defer() should be after Line: 400

Reply:
yes